src/unitTests/uid2ApiClient.test.ts

```diff
@@ -0,0 +1,167 @@
+import { NAME_CURVE, XhrMock, makeCstgOption, makeIdentityV2 } from '../mocks';
+import { Uid2ApiClient } from '../uid2ApiClient';
+import { base64ToBytes, bytesToBase64 } from '../uid2Base64';
+import { sdkWindow } from '../uid2Sdk';
+
+describe('UID2 API client tests', () => {
+  let uid2ApiClient: Uid2ApiClient;
+  let serverKeyPair: CryptoKeyPair;
+  let serverPublicKey: string;
+  let xhrMock: any;
+
+  beforeAll(async () => {
+    serverKeyPair = await crypto.subtle.generateKey(
+      {
+        name: 'ECDH',
+        namedCurve: NAME_CURVE,
+      },
+      false,
+      ['deriveKey']
+    );
+
+    const serverPubKeyArray = await crypto.subtle.exportKey('spki', serverKeyPair.publicKey);
+    serverPublicKey = `UID2-X-L-` + bytesToBase64(new Uint8Array(serverPubKeyArray));
+  });
+
+  beforeEach(() => {
+    uid2ApiClient = new Uid2ApiClient({});
+    xhrMock = new XhrMock(sdkWindow);
+  });
+
+  describe('#callCstgApi', () => {
+    const makeCstgApiCall = async () => {
+      return uid2ApiClient.callCstgApi(
+        { emailHash: 'lz3+Rj7IV4X1+Vr1ujkG7tstkxwk5pgkqJ6mXbpOgTs=' },
+        makeCstgOption({ serverPublicKey })
+      );
+    };
+
+    describe('When API returns 200', () => {
+      it('should decrypt identity from API response', async () => {
+        const uid2Token = makeIdentityV2();
+        xhrMock.send.mockImplementation((body: string) => {
+          const requestBody = JSON.parse(body);
+          xhrMock.sendEncryptedCSTGResponse(
+            {
+              clientPublicKey: base64ToBytes(requestBody.public_key),
+              serverPrivateKey: serverKeyPair.privateKey,
+            },
+            { status: 'success', body: uid2Token }
+          );
+        });
+
+        const cstgResult = await makeCstgApiCall();
+        expect(cstgResult.identity).toEqual(uid2Token);
+      });
+
+      it('should throw error when response has invalid response body', async () => {
+        xhrMock.send.mockImplementation((body: string) => {
+          const requestBody = JSON.parse(body);
+          xhrMock.sendEncryptedCSTGResponse(
+            {
+              clientPublicKey: base64ToBytes(requestBody.public_key),
+              serverPrivateKey: serverKeyPair.privateKey,
+            },
+            {}
+          );
+        });
+
+        try {
+          await uid2ApiClient.callCstgApi(
```

Comment:
If this doesn't throw, will the test pass?

Reply:
I've now updated try-catch to use `rejects` instead